demos/isoline_simplification/medial_axis_helpers.h

```diff
@@ -0,0 +1,104 @@
+/*
+The CartoCrow library implements algorithmic geo-visualization methods,
+developed at TU Eindhoven.
+Copyright (C) 2024 TU Eindhoven
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3f of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef CARTOCROW_MEDIAL_AXIS_HELPERS_H
+#define CARTOCROW_MEDIAL_AXIS_HELPERS_H
+
+#include <CGAL/Segment_Delaunay_graph_2.h>
+#include <CGAL/Segment_Delaunay_graph_adaptation_policies_2.h>
+#include <CGAL/Segment_Delaunay_graph_adaptation_traits_2.h>
+#include <CGAL/Segment_Delaunay_graph_hierarchy_2.h>
+#include <CGAL/Segment_Delaunay_graph_traits_2.h>
+
+template<class K,
+          class Gt  = CGAL::Segment_Delaunay_graph_filtered_traits_without_intersections_2<K, CGAL::Field_with_sqrt_tag>,
+          class SDG = CGAL::Segment_Delaunay_graph_hierarchy_2<Gt>,
+          class AT  = CGAL::Segment_Delaunay_graph_adaptation_traits_2<SDG>>
+bool same_points(const SDG& dg, const typename AT::Site_2& p, const typename AT::Site_2& q) {
+	return dg.geom_traits().equal_2_object()(p, q);
+}
+
+template<class K,
+          class Gt  = CGAL::Segment_Delaunay_graph_filtered_traits_without_intersections_2<K, CGAL::Field_with_sqrt_tag>,
+          class SDG = CGAL::Segment_Delaunay_graph_hierarchy_2<Gt>,
+          class AT  = CGAL::Segment_Delaunay_graph_adaptation_traits_2<SDG>>
+bool is_endpoint_of_segment(const SDG& dg, typename AT::Site_2& p, typename AT::Site_2& s) {
+	CGAL_precondition( p.is_point() && s.is_segment() );
```

Comment:
Is this code coming from / adapted from CGAL?

Reply:
Yes, good catch, I forgot this. I put the original license header at the top of the file. There were also a couple other functions I adapted that are now in a separate file with the original license header.